example/aws/cognito/README.md

```diff
@@ -0,0 +1,270 @@
+        
+        # Notebooks
+        kustomize build apps/jupyter/notebook-controller/upstream/overlays/kubeflow | kubectl apply -f -
+        kustomize build apps/jupyter/jupyter-web-app/upstream/overlays/istio | kubectl apply -f -
+        
+        # Admission Webhook
+        kustomize build apps/admission-webhook/upstream/overlays/cert-manager | kubectl apply -f -
+        
+        # Profiles + KFAM
+        kustomize build apps/profiles/upstream/overlays/kubeflow | kubectl apply -f -
+        
+        # Volumes Web App
+        kustomize build apps/volumes-web-app/upstream/overlays/istio | kubectl apply -f -
+        
+        # Tensorboard
+        kustomize build apps/tensorboard/tensorboards-web-app/upstream/overlays/istio | kubectl apply -f -
+        kustomize build apps/tensorboard/tensorboard-controller/upstream/overlays/kubeflow | kubectl apply -f -
+        
+        # TFJob Operator
+        kustomize build apps/tf-training/upstream/overlays/kubeflow | kubectl apply -f -
+        
+        # Pytorch Operator
+        kustomize build apps/pytorch-job/upstream/overlays/kubeflow | kubectl apply -f -
+        
+        # MPI Operator
+        kustomize build apps/mpi-job/upstream/overlays/kubeflow | kubectl apply -f -
+        
+        # MXNet Operator
+        kustomize build apps/mxnet-job/upstream/overlays/kubeflow | kubectl apply -f -
+        
+        # XGBoost Operator
+        kustomize build apps/xgboost-job/upstream/overlays/kubeflow | kubectl apply -f -
+
+        # Ingress
+        kustomize build distributions/aws/istio-ingress/overlays/cognito | kubectl apply -f -
+
+        # ALB controller
+        kustomize build distributions/aws/aws-alb-ingress-controller/base | kubectl apply -f -
+
+        # Envoy filter
+        kustomize build distributions/aws/aws-istio-envoy-filter/base | kubectl apply -f -
+        ```
+    2. Annotate the service account with the IAM role
+        1. ```
+            kubectl annotate serviceaccount -n kubeflow alb-ingress-controller eks.amazonaws.com/role-arn=$IAM_ROLE_ARN_FOR_IRSA
+            kubectl annotate serviceaccount -n kubeflow profiles-controller-service-account eks.amazonaws.com/role-arn=$IAM_ROLE_ARN_FOR_IRSA
+            ```
+    3. Check if ALB is provisioned. It takes around 3-5 minutes
+        1. ```
+            kubectl get ingress -n istio-system
+            Warning: extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
+            NAME            CLASS    HOSTS   ADDRESS                                                                  PORTS   AGE
+            istio-ingress   <none>   *       ebde55ee-istiosystem-istio-2af2-1100502020.us-west-2.elb.amazonaws.com   80      15d
+            ```
+        2. If `ADDRESS` is empty after a few minutes, check the logs of alb-ingress-controller by following [this guide](https://www.kubeflow.org/docs/distributions/aws/troubleshooting-aws/#alb-fails-to-provision)
```

Comment:
Address was empty because the node-group iam role on my eks cluster did not have the necessary permissions. Error message was `kubebuilder/controller "msg"="Reconciler error" "error"="failed to reconcile LB managed SecurityGroup: failed to reconcile managed LoadBalancer securityGroup: UnauthorizedOperation: You are not authorized to perform this operation.`  I gave it admin permissions however there may be some other permissions that is better ofc.

Reply:
fixed

Reply:
what was the fix?

Reply:
annotated the SA before deploying the controller